packer_test/common/check/file_gadgets.go

```diff
@@ -35,31 +35,24 @@ func FileExists(filePath string, isDir bool) Checker {
 	}
 }
 
-type fileInDir struct {
-	filename string
-	dirPath  string
+type fileGlob struct {
+	filepath string
 }
 
-func (fe fileInDir) Check(_, _ string, _ error) error {
-	files, err := os.ReadDir(fe.dirPath)
+func (fe fileGlob) Check(_, _ string, _ error) error {
+	matches, err := filepath.Glob(fe.filepath)
 	if err != nil {
-		return fmt.Errorf("failed to read dir %q: %s", fe.dirPath, err)
+		return fmt.Errorf("error finding file %q: %q", fe.filepath, err)
```

Comment:
This error feels misleading, `filepath.Glob` only fails if the glob is malformed, this error message should reflect that.

Reply:
To be honest I wonder if we shouldn't directly `panic` when invoking `FileGlob`, so that we don't even need to run the command before the checker fails to be produced.
The structure can keep a reference to the glob instance here, so we can fail early, and not when the command has already run.

In a test environment, even when there's a panic, the resources should be cleaned-up so this should be reasonably safe to call.

Reply:
The key here is that the files are only generated after running the `packer build` command, which happens between initializing `FileGlob` and running `Check`. Therefore, `FileGlob` just stores the pattern, and `Check` performs the actual file match afterward, once files exist. 

If we want to enforce an initial syntax check, we could indeed call `filepath.Glob` twice: once to validate the pattern immediately in FileGlob, and again in Check to confirm the files were generated. However, since this is primarily for testing, I’d personally skip the double check to keep things simple.

Let me know if you’d like to discuss further, though!